### PR TITLE
Cloud Spanner Lock Provider

### PR DESCRIPTION
**Pull request by pXius**
This PR is to add a [Google Cloud Spanner ](https://cloud.google.com/spanner?hl=en) lock provider for Shedlock.

This implementation respects Spanner's [transactionality](https://cloud.google.com/spanner/docs/transactions) by making use of read and write mutations within a transactional context.

The implementation is based on the `StorageBasedLockProvider` and accessor. 
Tests extend `AbstractStorageBasedLockProviderIntegrationTest` and make use of the Spanner emulator in a test container to test integration.

I've also included an optional `Configuration` class when creating the lock provider which can be used for a custom host name, table and column names. 

Finally, I was unsure if I was supposed to update the readme and parent pom with the new module. I did so anyway, but I'm of course open to change that and anything else as per your requirements.

### CODE REVIEW

**Comment by pXius** [comment]
Having issues with `shouldCreateTwoIndependentLocks` test that I'm unable to reproduce locally. 

Will see what I can do.

**Comment by lukas-krecan** [comment]
Hi, thanks a lot. I will look at the PR soon

File: providers/spanner/shedlock-provider-spanner/pom.xml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-jar-plugin</artifactId>
+                <configuration>
+                    <archive>
+                        <manifestEntries>
+                            <Automatic-Module-Name>
+                                net.javacrumbs.shedlock.provider.mongo
```
Comment (lukas-krecan) [inline providers/spanner/shedlock-provider-spanner/pom.xml:67]: Please change this

**Comment by lukas-krecan** [comment]
Hi, I have taken the liberty to do some small refactorings [here](https://github.com/lukas-krecan/ShedLock/pull/1592/commits/86042de73e1d43bdf4deb8bf2d631791c9225594). If you are OK with the changes, I will merge it and release it. Thanks again.

**Comment by pXius** [comment]
Looks great to me!
Thanks for the swift feedback and attentiveness.

**Comment by lukas-krecan** [comment]
Release as 5.10.0

**Pull request merged**